ext/standard/math.c

```diff
@@ -93,19 +93,19 @@ static inline double php_intpow10(int power) {
 /* {{{ php_round_helper
        Actually performs the rounding of a value to integer in a certain mode */
 static inline double php_round_helper(double value, int mode) {
-	double integral, fractional;
 
 	/* Split the input value into the integral and fractional part.
 	 *
 	 * Both parts will have the same sign as the input value. We take
 	 * the absolute value of the fractional part (which will not result
 	 * in branches in the assembly) to make the following cases simpler.
 	 */
-	fractional = fabs(modf(value, &integral));
+	double integral;
+	double fractional = modf(value, &integral);
```

Comment:
I don't see any case for the new rounding modes where the sign of `fractional` would be important, thus please revert this to the original implementation with `fabs()`.

Reply:
I reused the `fabs` function for the previous modes, however I didn't add them to the new ones. Do you think this is good approach?

Reply:
The previous implementation was specifically written like that to generate good assembly with consistent performance, which I verified and benchmarked. Calling `fabs()` is super cheap, because it's just some (simple) bit operations and it importantly does not require any assembly-level branches (which are expensive).

By moving the `fabs()` call into each `case` separately, the code will be harder to optimize by the compiler and I expect the code size to increase, putting additional pressure on the instruction cache. Personally I also find it harder to reason about the code, because `fractional` is no longer guaranteed to be non-negative.

Reply:
Thank you for the clear explanation. I understand now the reason and reverted back the change. 🙂